src/plugins/dna_export/src/ExportTasks.cpp

```diff
@@ -179,6 +193,9 @@ void ExportMSA2MSATask::run() {
             aminoTT->translate(seq.constData(), seq.length(), resseq.data(), resseq.length());
 
             resseq.replace("*", "X");
+            if (!trimGaps && convertUnknownToGap) {
+                resseq.replace("X", "-");
```

Comment:
This is still not correct. 

Here is 2 reasons why:
1) We can't replace all X with gaps, because you replace stop codons (valid nucleic triplets, non-gaps) with gaps.
The simplest solution I see here is to move "resseq.replace("*", "X");" below your replacement code

2) There is a chance that some incomplete DNA triplets are always converted to the same amino. 
For example: Proline, Serine, etc... may have "any" base in the end position and still be translated into a valid protein.
https://miro.medium.com/max/1440/0*YVhxcZ55DqPBDdGF.png

This way if in the original DNA alignment 'gap' means 'unknown', the translation still should contain a valid amino acid, not a gap. 

I'm OK if the item 2) is fixed as a separate bug/PR.

Reply:
1. Fixed
2. I understand the idea, but, actually, don't get a sequence, which could be a part of alignment and affected by this problem. Could you help me?

Reply:
Ok, I will create a separate issue.